indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -1108,13 +1108,12 @@ private static InputFormat getInputFormat(IndexIngestionSpec ingestionSchema)
   }
 
   @Override
-  public void cleanUp(TaskToolbox toolbox, @Nullable TaskStatus taskStatus) throws Exception
+  protected boolean shouldCleanupTask()
```

Comment:
Is this change really needed?
Is `shouldCleanupTask()` meant to be used in implementations other than `IndexTask`?

The javadoc of `Task.cleanup()` already calls out that this method is overridden only by implementations that need to perform any special cleanup. If not overridden, the default impl of `cleanup` is noop.
Adding another `shouldCleanupTask()` method in the mix seems unnecessary, esp since it is used in just one place.

Reply:
shouldCleanupTask is used in ParallelIndexSupervisorTask as well. i thought rather than copying the cleanupLogic in IndexTask (check the boolean and call super.cleanup() if true), it would be easier to add a new method that could be implemented, but i think what you say makes sense, i can change it back.